017-strimzi-ui-mono-repo.md

```diff
@@ -0,0 +1,53 @@
+
+# Converting Strimzi UI Mono Repo
+
+The following proposal is a recommendation that we move to a mono repo and separate the current UI into smaller packages.
+
+## Current situation
+
+Currently the [Strimzi UI](./011-strimzi-ui.md) is being developed as one monolithic application for the client and server.  In order to help isolate code into manageable packages it is recommended that the project moves to a mono repo with smaller packages. This architecture will help clearly define functional areas and help with separation of concerns within the UI client and server. We will continue to have one single source of truth for the UI by having a monorepo, however with smaller packages it will allow UIs to pull in only the packages that are needed to build their UI without additional dependencies.  
+
+## Motivation
+
+By moving to a mono repo the project will gain the following:
+ * **Simplified dependency management for view layers:**  Each of the view layers can pull in the functionality that the view needs by including only the packages the are required for that view.
+ 
+ * **Focus on immediate client applications needs:**  The view layer makes up what the user will interact with for each application being built.  Each applications views are swapped out at build time via webpack normalization. This process requires at minimum skelton code for each of the view layers be added in order for the client to build. Short term this can result in extra code being generated and additional rework in the future based on each of the clients immediate needs for their consumers.  In order to reduce this rework and allow each client to develop views as needed, it is recommended that view layers be stored in a separate package (e.g. ui-carbon, ui-patternfly).
```

Comment:
Kafka is full of clients and consumers. What does the client mean in this context? Can you clarify what does it mean.

Reply:
meant to be web client will update the wording through this document.